lib/api/model/initial_snapshot.dart

```diff
@@ -259,7 +259,7 @@ class UnreadDmSnapshot {
 
   UnreadDmSnapshot({
     required this.otherUserId,
-    required this.unreadMessageIds,
+    required this.unreadMessageIds, // TODO assert sorted?
```

Comment:
I think I was originally thinking a check that would actually throw in production, since it'll always be a problem if it happens. Like we throw in production when e.g. mark-as-unread events are missing a `message_details` property:

```dart
  factory UpdateMessageFlagsRemoveEvent.fromJson(Map<String, dynamic> json) {
    final result = _$UpdateMessageFlagsRemoveEventFromJson(json);
    // Crunchy-shell validation
    if (
      result.flag == MessageFlag.read
      && true // (we assume `event_types` has `message` and `update_message_flags`)
    ) {
      result.messageDetails as Map<int, UpdateMessageFlagsMessageDetail>;
    }
    return result;
  }
```

Reply:
Unless I hear otherwise I'll make it a regular `assert` in my next revision, but I'll add tests that the `assert` is there, which we can easily adapt to logic that throws in production (in which case tests will be necessary)

Reply:
> I think I was originally thinking a check that would actually throw in production, since it'll always be a problem if it happens.

Yeah, the reason I hesitate to add a check like that is that it's not free — it's a linear-time scan through each of these lists, so scanning through 50k elements at initial-fetch time in the case of a user with lots of unreads. That's something we could certainly afford if we had a good reason we needed it (we're unavoidably doing linear work on this data to process it, and this would be cheaper than other things we do); but since the server already gives us a guarantee, better not to spend time cross-checking it.